rust/blockstore/src/types/writer.rs

```diff
@@ -104,4 +104,14 @@ impl BlockfileWriter {
             BlockfileWriter::ArrowOrderedBlockfileWriter(writer) => writer.id(),
         }
     }
+
+    pub fn forked_from_id(&self) -> Option<uuid::Uuid> {
```

Comment:
given a writer, this allows us to construct a reader for the previous blockfile version